drizzlepac/haputils/catalog_utils.py

```diff
@@ -197,69 +210,99 @@ def compute_background(self, box_size, win_size,
 
         # SExtractorBackground ans StdBackgroundRMS are the defaults
         bkg = None
+        is_zero_background_defined = False
 
         # Make a local copy of the data(image) being processed in order to reset any
         # data values which equal nan (e.g., subarrays) to zero.
         imgdata = self.data.copy()
-        imgdata[np.isnan(imgdata)] = 0
-
-        # Create mask to reject any sources located less than 10 pixels from a image/chip edge
-        # to be used for the sigma-clipping
-        binary_inverted_wht = np.where(imgdata == 0, 1, 0)
-        self.exclude_zones_mask = ndimage.binary_dilation(binary_inverted_wht, iterations=10)
-
-        # Compute a sigma-clipped background which returns only single values for mean,
+        fits.PrimaryHDU(data=imgdata).writeto(self.imgname.replace(".fits", '_imgdata.fits'))
+
+        # In order to compute the proper statistics on the input data, need to use the footprint
+        # mask to get the actual data - illuminated portion (True), non-illuminated (False).
+        self.footprint_mask = self.num_images_mask > 0
+        self.inv_footprint_mask = np.invert(self.footprint_mask)
+
+        # If the image contains a lot of values identically equal to zero (as in some SBC images),
+        # set the two-dimensional background image to a constant of zero and the background rms to
+        # the real rms of the non-zero values in the image.
+        no_of_illuminated_pixels = self.footprint_mask.sum()
+        no_of_zeros = np.count_nonzero(imgdata[self.footprint_mask] == 0)
+        non_zero_pixels = imgdata[self.footprint_mask] 
+
+        # BACKGROUND COMPUTATION 1 (unusual case)
+        # If there are too many background zeros in the image (> 25%), set the
+        # background median and background rms values
+        if no_of_zeros / float(no_of_illuminated_pixels) > 0.25: 
```

Comment:
This 'magic' number of 25% (0.25) should be exposed as a parameter in the configuration json files in the pars directory.